packages/pushnotification/src/PushNotification.ts

```diff
@@ -186,87 +193,106 @@ export default class PushNotification {
 		this._currentState = nextAppState;
 	}
 
-	handleCampaignPush(rawMessage) {
-		let message = rawMessage;
-		let campaign = null;
+	parseMessageData = rawMessage => {
+		let eventSource = null;
+		let eventSourceAttributes = {};
+
 		if (Platform.OS === 'ios') {
-			message = this.parseMessageFromIOS(rawMessage);
-			campaign =
-				message && message.data && message.data.pinpoint
-					? message.data.pinpoint.campaign
-					: null;
+			const message = this.parseMessageFromIOS(rawMessage);
+			const pinpointData =
+				message && message.data ? message.data.pinpoint : null;
+			if (pinpointData.campaign) {
+				eventSource = 'campaign';
+				eventSourceAttributes = pinpointData.campaign;
+			} else if (pinpointData.journey) {
+				eventSource = 'journey';
+				eventSourceAttributes = pinpointData.journey;
+			}
```

Comment:
Adding some additional checks to make sure `pinpointData` object exists.

```suggestion
			const pinpointData =
				message && message.data ? message.data.pinpoint : null;
			if (pinpointData && pinpointData.campaign) {
				eventSource = 'campaign';
				eventSourceAttributes = pinpointData.campaign;
			} else if (pinpointData && pinpointData.journey) {
				eventSource = 'journey';
				eventSourceAttributes = pinpointData.journey;
			}
```

Reply:
Good point. Updated.